core/rio/api/src/main/java/org/eclipse/rdf4j/rio/RDFFormat.java

```diff
@@ -264,14 +264,22 @@
 				qValue -= 1;
 			}
 
-			for (String mimeType : format.getMIMETypes()) {
-				String acceptParam = mimeType;
+			if (RDFXML.equals(format)) {
+				// We explicitly dislike RDF/XML as it has limitations in what it can serialize. See #299.
+				qValue -= 4;
+			}
 
-				if (qValue < 10) {
-					acceptParam += ";q=0." + qValue;
-				}
+			// if the qValue did not go negative, we add this format to the accept params.
+			if (qValue > 0) { 
```

Comment:
I would prefer to have a minimum value rather than excluding them. If there are issues on the other side with low q-value formats being selected then I don't think it will be fixed by eliminating some that users know are on their classpath.